### PR TITLE
Should speed up ASV benchmarks significantly

### PR DESCRIPTION
**Pull request by IvoDD**
- Separates non-batch and batch BasicFunctions. Uses multiple symbols only where it is required
- Use single symbol for ModificationFunctions
- Use less rows for batch benchmarks
- Decrease version chain length for IteratingVersionChain

### CODE REVIEW

File: python/benchmarks/basic_functions.py [hunk 1]
```diff
@@ -78,20 +77,87 @@ def get_fresh_lib(self):
...
+
+        self.df = generate_pseudo_random_dataframe(rows)
+        self.lib = self.ac[get_prewritten_lib_name(rows)]
+        self.fresh_lib = self.get_fresh_lib()
+
+    def get_fresh_lib(self):
+        self.ac.delete_library("fresh_lib")
+        return self.ac.create_library("fresh_lib")
+
     def time_write_staged(self, rows, num_symbols):
```
Comment (alexowens90) [inline python/benchmarks/basic_functions.py:161]: Staged writes aren't a batch method, move to `BasicFunctions` and make sure there are multiple incomplete segments to compact

File: python/benchmarks/basic_functions.py [hunk 2]
```diff
@@ -78,20 +77,87 @@ def get_fresh_lib(self):
...
+
+    def peakmem_read_with_date_ranges(self, rows):
+        self.lib.read(f"sym", date_range=BasicFunctions.DATE_RANGE).data
+
+
+class BatchBasicFunctions:
+    number = 5
+    timeout = 6000
+    CONNECTION_STRING = "lmdb://batch_basic_functions?map_size=20GB"
+    DATE_RANGE = pd.date_range("2023-01-01", "2023-01-01")
```
Comment (alexowens90) [inline python/benchmarks/basic_functions.py:125]: I know this hasn't changed, but this date range will filter out all rows, which I'm guessing isn't the intended behaviour

**Review by alexowens90 — CHANGES_REQUESTED** [review]

File: python/benchmarks/basic_functions.py (hunk 1 above)
Comment (IvoDD) [reply python/benchmarks/basic_functions.py]: Agreed. Thanks for flagging!
Should be fixed now.

File: python/benchmarks/basic_functions.py (hunk 2 above)
Comment (IvoDD) [reply python/benchmarks/basic_functions.py]: Thanks, moved it to `2022-12-31, 2023-01-01`, which should be non empty (last timestamp on all dataframes should be 2023-01-01)

**Review by alexowens90 — APPROVED** [review]

**Pull request merged**